File: Easy/2025-09-13-3541-Find-Most-Frequent-Vowel-and-Consonant/solution.js
var maxVowelsAndConsonants = function(s) {
    const vowelSet = new Set(['a', 'e', 'i', 'o', 'u']);
    const frequency = {};
    
    for (const char of s) {
        frequency[char] = (frequency[char] || 0) + 1;
    }
    
    let maxVowelFreq = 0;
    let maxConsonantFreq = 0;
    
    for (const char in frequency) {
        if (vowelSet.has(char)) {
            maxVowelFreq = Math.max(maxVowelFreq, frequency[char]);
        } else {
            maxConsonantFreq = Math.max(maxConsonantFreq, frequency[char]);
        }
    }
    
    return maxVowelFreq + maxConsonantFreq;
};

File: Easy/2025-09-13-3541-Find-Most-Frequent-Vowel-and-Consonant/solution.java
class Solution {
    public int maxVowelsAndConsonants(String s) {
        int[] freq = new int[26];
        for (char c : s.toCharArray()) {
            freq[c - 'a']++;
        }

        int maxVowelFreq = 0;
        int maxConsonantFreq = 0;

        for (char c = 'a'; c <= 'z'; c++) {
            if ("aeiou".indexOf(c) != -1) {
                maxVowelFreq = Math.max(maxVowelFreq, freq[c - 'a']);
            } else {
                maxConsonantFreq = Math.max(maxConsonantFreq, freq[c - 'a']);
            }
        }

        return maxVowelFreq + maxConsonantFreq;
    }
}